### PR TITLE
[rpcchainvm] use SIGUSR1 as shutdown signal for subprocess

### PR DESCRIPTION
**Pull request by hexfusion**
## Why this should be merged

While replacing go-plugin in an attempt to provide better graceful termination support we included a regression for systemd managed services. Prior to go-plugin removal the `SIGTERM` signal was not honored by the plugin process. systemd's default KillMode is `control-group` which on stop will send `SIGTERM` to all processes in the cgroup. While this is expected in AvalancheGo it is not expected by the VM process.

## How this works

Avoid systemd's default KillMode intent by only logging `SIGTERM` signals and replacing `SIGTERM` with `SIGUSR1` as the expected termination signal.

## How this was tested

Currently untested outside of Ci

### CODE REVIEW

File: vms/rpcchainvm/linux_vm.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+					return
+				}
+			case <-ctx.Done():
+				fmt.Println("runtime engine: context has been cancelled")
+				return
+			}
+		}
+	}(ctx)
+
+	// address of Runtime server from ENV
```
Comment (hexfusion) [inline vms/rpcchainvm/linux_vm.go:58]: deduping this logic

File: vms/rpcchainvm/non_linux_vm.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"os/signal"
+	"syscall"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/vms/rpcchainvm/grpcutils"
+)
+
+func serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption) error {
+	signals := make(chan os.Signal, 2)
+	signal.Notify(signals, syscall.SIGKILL)
```
[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply vms/rpcchainvm/non_linux_vm.go:22]: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170#remarks

File: vms/rpcchainvm/linux_vm.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+//go:build linux
```
[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply vms/rpcchainvm/linux_vm.go:4]: Looks to be supported: https://github.com/golang/go/blob/4c5130a96eabd5d9a72a43aa8e895b668fbd653b/src/runtime/defs_darwin_arm64.go#L63

File: vms/rpcchainvm/non_linux_vm.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply vms/rpcchainvm/non_linux_vm.go:22]: I think SIGKILL is the right way to terminate still

File: vms/rpcchainvm/non_unix_vm.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		for {
+			select {
+			case s := <-signals:
+				switch s {
+				case syscall.SIGINT, syscall.SIGTERM:
+					// these signals are ignored because ctrl-c will send SIGINT
+					// to all processes and systemd will send SIGTERM to all
+					// processes by default during stop.
+					fmt.Printf("runtime engine: ignoring signal: %s\n", s)
+				case syscall.SIGUSR1:
```
Comment (patrick-ogrady) [inline vms/rpcchainvm/non_unix_vm.go:38]: I think this should be SIGKILL?

File: vms/rpcchainvm/unix_vm.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	"os/signal"
+	"syscall"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/vms/rpcchainvm/grpcutils"
+)
+
+func serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption) error {
+	signals := make(chan os.Signal, 2)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR1)
```
Comment (patrick-ogrady) [inline vms/rpcchainvm/unix_vm.go:24]: SIGUSR2

File: vms/rpcchainvm/unix_vm.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/vms/rpcchainvm/grpcutils"
+)
+
+func serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption) error {
+	signals := make(chan os.Signal, 2)
```
Comment (patrick-ogrady) [inline vms/rpcchainvm/unix_vm.go:23]: Should this now be 3?

**Pull request closed without merging**